hadoop-client-modules/hadoop-client-runtime/pom.xml

```diff
@@ -155,6 +155,8 @@
                       <exclude>org.ow2.asm:*</exclude>
                       <!-- Leave bouncycastle unshaded because it's signed with a special Oracle certificate so it can be a custom JCE security provider -->
                       <exclude>org.bouncycastle:*</exclude>
+                      <!-- Leave snappy that includes native methods which cannot be relocated. -->
+                      <exclude>org.xerial.snappy:*</exclude>
```

Comment:
hmm I'm wondering if snappy should be a runtime dependency in this module - currently it is compile inherited from hadoop-client-api

Reply:
`hadoop-client-api` is already runtime dependency here. Do we still need to add snappy-java here?

Reply:
When I check dependency:tree, snappy is compile scope:
```
[INFO] |  +- org.apache.hadoop:hadoop-mapreduce-client-core:jar:3.4.0-SNAPSHOT:compile (optional)·
[INFO] |  |  \- org.apache.hadoop:hadoop-yarn-common:jar:3.4.0-SNAPSHOT:compile (optional)·
[INFO] |  |     +- javax.servlet:javax.servlet-api:jar:3.1.0:compile (optional)·
[INFO] |  |     +- com.sun.jersey:jersey-core:jar:1.19:compile (optional)·
[INFO] |  |     |  \- javax.ws.rs:jsr311-api:jar:1.1.1:compile (optional)·
[INFO] |  |     +- com.sun.jersey:jersey-client:jar:1.19:compile (optional)·
[INFO] |  |     +- com.fasterxml.jackson.module:jackson-module-jaxb-annotations:jar:2.10.5:compile (optional)·
[INFO] |  |     |  \- jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:compile (optional)·
[INFO] |  |     \- com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:jar:2.10.5:compile (optional)·
[INFO] |  |        \- com.fasterxml.jackson.jaxrs:jackson-jaxrs-base:jar:2.10.5:compile (optional)·
[INFO] |  \- org.apache.hadoop:hadoop-mapreduce-client-jobclient:jar:3.4.0-SNAPSHOT:compile (optional)·
[INFO] |     \- org.apache.hadoop:hadoop-mapreduce-client-common:jar:3.4.0-SNAPSHOT:compile (optional)·
[INFO] +- org.apache.hadoop:hadoop-client-api:jar:3.4.0-SNAPSHOT:runtime
[INFO] |  \- org.xerial.snappy:snappy-java:jar:1.1.8.2:compile
[INFO] +- org.apache.hadoop:hadoop-annotations:jar:3.4.0-SNAPSHOT:compile
[INFO] +- org.slf4j:slf4j-api:jar:1.7.30:runtime
[INFO] +- commons-logging:commons-logging:jar:1.1.3:runtime
[INFO] +- com.google.code.findbugs:jsr305:jar:3.0.2:runtime
[INFO] \- log4j:log4j:jar:1.2.17:runtime (optional)·
```

Reply:
Okay. So not like `optional` which seems can be inherited, we need to make it runtime explicitly.